src/appleseed.shared/application/application.cpp

```diff
@@ -137,6 +137,44 @@ const char* Application::get_root_path()
     return root_path_buffer;
 }
 
+const char* Application::get_user_settings_path()
+{
+    static char user_settings_buffer[FOUNDATION_MAX_PATH_LENGTH + 1];
+    static bool user_settings_initialized = false;
+
+    if (!user_settings_initialized)
+    {
+// Windows.
+#if defined _WIN32
+
+        return 0;
+
+// OS X.
+#elif defined __APPLE__
+
+        return 0;
+
+#elif defined __linux__
```

Comment:
Correct, both GNU/Linux, FreeBSD (all *BSDs, FWIW), and other Unix-like systems behave the same WRT storing user's settings in `$HOME`.

I cannot be 100% sure about OSX, but likely there as well (OSX often behaves like Unix) — or does the user have write permissions to `foobar.app` bundle installation directory?